helpers/target.go

```diff
@@ -129,3 +154,332 @@ func isAllowed(addr string) (bool, error) {
 	}
 	return true, nil
 }
+
+// ServiceCreds represents the credentials
+// necessary to access an authenticated service.
+// There are constructors available in this same
+// package for:
+//    - AWS Assume role through vulcan-assume-role svc.
+//    - Docker registry.
+//    - Github repository.
+type ServiceCreds interface {
+	URL() string
+	Username() string
+	Password() string
+}
+
+// AWSCreds holds data required
+// to perform an assume role request.
+type AWSCreds struct {
+	AssumeRoleURL string
+	Role          string
+}
+
+// NewAWSCreds creates a new AWS Credentials for Assume Role.
+func NewAWSCreds(assumeRoleURL, role string) *AWSCreds {
+	return &AWSCreds{
+		AssumeRoleURL: assumeRoleURL,
+		Role:          role,
+	}
+}
+func (c *AWSCreds) URL() string {
+	return c.AssumeRoleURL
+}
+func (c *AWSCreds) Username() string {
+	return c.Role
+}
+func (c *AWSCreds) Password() string {
+	return ""
+}
+
+type DockerCreds struct {
+	RegistryURL string
+	User        string
+	Pass        string
+}
+
+// DockerHubCreds represents a void
+// DockerCreds struct allowed to be
+// used with Docker Hub registry.
+var DockerHubCreds = &DockerCreds{}
+
+// NewDockerCreds creates a new Docker Credentials struct.
+func NewDockerCreds(registryURL, user, pass string) *DockerCreds {
+	return &DockerCreds{
+		RegistryURL: registryURL,
+		User:        user,
+		Pass:        pass,
+	}
+}
+func (c *DockerCreds) URL() string {
+	return c.RegistryURL
+}
+func (c *DockerCreds) Username() string {
+	return c.User
+}
+func (c *DockerCreds) Password() string {
+	return c.Pass
+}
+
+type GitCreds struct {
+	RepoURL string
+	User    string
+	Pass    string
+}
+
+// NewGitCreds creates a new Git Credentials struct.
+// user and pass can be void if no auth is required.
+func NewGitCreds(repoURL, user, pass string) *GitCreds {
+	return &GitCreds{
+		RepoURL: repoURL,
+		User:    user,
+		Pass:    pass,
+	}
+}
+func (c *GitCreds) URL() string {
+	return c.RepoURL
+}
+func (c *GitCreds) Username() string {
+	return c.User
+}
+func (c *GitCreds) Password() string {
+	return c.Pass
+}
+
+// IsReachable returns wether target is reachable
+// so the check execution can be performed.
+// If asset type is void, it will be inferred from target.
+//
+// ServiceCredentials are required for AWS, Docker and Git types.
+// Constructors for AWS, Docker and Git credentials can be found
+// in this same package.
+//
+// Verifications made depend on the asset type:
+//    - IP: None.
+//    - IPRange: None.
+//    - DomainName: None.
```

Comment:
I feel that somehow this `IsReachable` should include the `IsScannable` premises, like knowing if the IPs are reachable or the domain is really a domain (SOA record). But let's discuss this further.

Reply:
So if I understood you right, you are saying that we should also verify in the `IsReachable` function if the IPs are private or not? I'm not sure about that because we are already doing that in the `IsScannable` function in a common place in the SDK for all the checks, so we would be repeting that verification, no? Also, if we do that on the check level, calling the 'IsReachable` function, at that point I believe we are not aware of the check configuration, so we would not know if private IPs should be allowed no?

This is tricky, let me know if I understood you correctly.

Regarding the Domain, so what would be the verification here? Basically using the `types.IsDomainName` function no?

Reply:
Added domain verification.